### PR TITLE
Fix several bugs with external libraries

### PR DESCRIPTION
**Pull request by nbirillo**

### CODE REVIEW

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/PluginConfig.kt [hunk 1]
```diff
@@ -67,7 +65,7 @@ class PluginConfig(
      *
      * @return [StringBuilder]
```
Comment (CommanderTvis) [inline reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/PluginConfig.kt:66]: Not a `StringBuilder` actually.

Comment (nbirillo) [reply reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/PluginConfig.kt:66]: it is diktat with auto generation, sorry((

File: README.md [hunk 2]
```diff
@@ -103,16 +103,30 @@ Thirdly, customize the Reflekt plugin. In the `build.gradle.kts` file, add the `
...
-    // Use only DependencyHandlers which have canBeResolve = True
-    // Note: Reflekt works only with kt files from libraries
-    librariesToIntrospect = listOf("io.kotless:kotless-dsl:0.1.6")
 }
 ```
 
-_Please note that the `librariesToIntrospect` argument should contain only the dependencies that you
-use in the `dependencies` section. These dependencies may be implemented in Java or Kotlin language,
-but the analysis will be made only on Kotlin files._
+**Please note** that the Reflekt can also analyze the files from the external libraries.
```
Comment (elena-lyulina) [inline README.md:109]: that the Reflekt can -> that Reflekt can

File: README.md [hunk 3]
```diff
@@ -103,16 +103,30 @@ Thirdly, customize the Reflekt plugin. In the `build.gradle.kts` file, add the `
...
 
-_Please note that the `librariesToIntrospect` argument should contain only the dependencies that you
-use in the `dependencies` section. These dependencies may be implemented in Java or Kotlin language,
-but the analysis will be made only on Kotlin files._
+**Please note** that the Reflekt can also analyze the files from the external libraries. 
+Reflekt can handle only libraries from the `dependencies` section and 
+DependencyHandlers of them should be `canBeResolve = True`, e.g.:
+```kotlin
+val reflektConfiguration by configurations.creating {
+    isCanBeResolved = true
```
Comment (elena-lyulina) [inline README.md:114]: is it `canBeResolve = True` or `isCanBeResolved = true`?

File: README.md [hunk 4]
```diff
@@ -103,16 +103,30 @@ Thirdly, customize the Reflekt plugin. In the `build.gradle.kts` file, add the `
...
+}
+
+configurations["implementation"].extendsFrom(reflektConfiguration)
+```
+Also, this library should contain a special file `ReflektMeta`. The last point means the library
+should use Reflekt with the following configuration:
```
Comment (elena-lyulina) [inline README.md:120]: Also, this library should contain a specific file called `ReflektMeta` to store additional meta information. To enable it, the library should use Reflekt with the following configuration:

File: gradle-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/ReflektSubPlugin.kt [hunk 5]
```diff
@@ -47,16 +49,28 @@ class ReflektSubPlugin : KotlinCompilerPluginSupportPlugin {
...
-            librariesToIntrospect + reflektMetaFilesOptions + dependencyJars +
+            reflektMetaFilesFromLibrariesOptions + dependencyJars +
                 SubpluginOption(key = ENABLED_OPTION_INFO.name, value = extension.enabled.toString()) +
                 SubpluginOption(key = OUTPUT_DIR_OPTION_INFO.name, value = generationPath) +
                 SubpluginOption(key = SAVE_METADATA_OPTION_INFO.name, value = extension.toSaveMetadata.toString()) +
                 SubpluginOption(key = REFLEKT_META_FILE_PATH.name, value = createReflektMeta(project.getResourcesPath()).absolutePath)
         }
     }
 
+    // TODO: should we scan all dependencies or the user can set up it?
```
Comment (elena-lyulina) [inline gradle-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/ReflektSubPlugin.kt:63]: set up it -> set it up

File: gradle-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/ReflektSubPlugin.kt [hunk 6]
```diff
@@ -47,16 +49,28 @@ class ReflektSubPlugin : KotlinCompilerPluginSupportPlugin {
...
+    private fun Project.getReflektMetaFilesFromLibraries(): MutableSet<File> {
+        val reflektMetaFilesFromLibraries: MutableSet<File> = HashSet()
+        // We should miss all kotlinCompiler configuration since they wiil be resolved later by the Kotlin compiler
```
Comment (elena-lyulina) [inline gradle-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/ReflektSubPlugin.kt:66]: miss -> skip

**Review by elena-lyulina — APPROVED** [review]
Just a few small comments mostly about english, looks great, thanks ❤️

File: README.md (hunk 3 above)
Comment (CommanderTvis) [reply README.md:114]: No, that's Gradle's method.

**Pull request merged**